[elfio] Add new vcpkg port elfio

schultetwin1 opened the pull request:
Adds a new new port the elfio library

Fixes #14313

Commit:
[elfio] Add new vcpkg port elfio

Link1J commented:
I wrote one for this lib myself, as I planned to use it. Funny how someone already did it.
I just have one thing to ask, Is there any reason you used a usage file over using `vcpkg_fixup_cmake_targets` in the portfile?

Commit:
Use `vcpkg_fixup_cmake_targets` to fix install

elfio installs config files to path not supported by vcpkg. This uses
vcpkg_fixup_cmake_targets to fix up those paths.

schultetwin1 commented:
> Is there any reason you used a usage file over using vcpkg_fixup_cmake_targets in the portfile?

Only that I did not know it existed :) Thanks for the pointer!

NancyLi1013 commented on ports/elfio/portfile.cmake:
```diff
@@ -0,0 +1,22 @@
+
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(
+    CONFIG_PATH
+        share/${PORT}/cmake
+    TARGET_PATH
+        share/${PORT})
```
```suggestion
vcpkg_fixup_cmake_targets(CONFIG_PATH share/${PORT}/cmake TARGET_PATH share/${PORT})
```

NancyLi1013 commented on ports/elfio/portfile.cmake:
```diff
@@ -0,0 +1,22 @@
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+
```
```suggestion
```

schultetwin1 committed:
Apply suggestions from code review

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

NancyLi1013 commented:
LGTM, thanks for your PR.

BillyONeal committed:
Remove vcpkg_fixup_cmake_targets default TARGET_PATH.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal commented:
Thanks for your help!

schultetwin1 commented:
> Thanks for your help!

Thanks for your changes as well!
